pkg/archive/archive.go

```diff
@@ -530,7 +550,8 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 		}
 
 	case tar.TypeLink:
-		targetPath := filepath.Join(extractDir, hdr.Linkname)
+		lpath := adjustHardLink(hdr.Linkname, hdr.Name)
+		targetPath := filepath.Join(extractDir, lpath)
```

Comment:
Is `Replace()` safe to use here? Looks like it may replace matches in the middle of a path as well (thinking `HasPrefix()` + `TrimPrefix()` to prevent that case)